streams/shared/src/main/scala/zio/stream/ZPipeline.scala

```diff
@@ -1564,20 +1564,24 @@ object ZPipeline extends ZPipelinePlatformSpecificConstructors {
    * Encode each input byte as two output bytes as the hex representation of the
    * input byte.
    */
-  def hexEncode(implicit trace: Trace): ZPipeline[Any, Nothing, Byte, Char] = {
-    val DIGITS: Array[Char] = Array(
+  def hexEncode(upperCase: Boolean = false)(implicit trace: Trace): ZPipeline[Any, Nothing, Byte, Char] = {
```

Comment:
Maybe we just have one and the user can always use another operator if they want to convert the characters to upper if they want to?

Reply:
I'll revert the addition of the upperCase flag.

Also, I looked in to the use of `UnmappableCharacterException` and I don't think its a good match for the need. I want to be able to put a meaningful message and it overrides getMessage to just be "Input length = XXX" where XXX is from the constructor.